### DISCUSSION TITLE
CURD - Read working, Delete NOT

### DISCUSSION THREAD

**Discussion by saroar** [pointfreeco/vapor-routing#17] [General]
here is my Router

```swift

// USER
public enum UserRoute: Equatable {
    case find
    case deleteSoft
    case restore
    case delete
    
    
    case attachments(AttachmentsRoute)
    case conversations(ConversationsRoute)
    case devices(DevicesRoute)
    case events(EventsRoute)
    
}

public let userRouter = OneOf {
    Route(.case(UserRoute.find))
    
    Route(.case(UserRoute.deleteSoft)) {
        Method.delete
        Path { "soft" }
    }
    
    Route(.case(UserRoute.restore)) {
        Method.put
        Path { "restore" }
    }
    
    Route(.case(UserRoute.delete)) {
        Method.delete
    }
    
    Route(.case(UserRoute.attachments)) {
        Path { "attachments" }
        attachmentsRouter
    }
    
    Route(.case(UserRoute.conversations)) {
        Path { "conversations" }
        conversationsRouter
    }
    
    Route(.case(UserRoute.devices)) {
        Path { From(.utf8) { "devices".utf8 } }
        devicesRouter
    }
    
    Route(.case(UserRoute.events)) {
        Path { "events" }
        eventsRouter
    }
}

// USERS
public enum UsersRoute: Equatable {
    case user(id: String, route: UserRoute)
    case update(input: UserOutput)
}

public let usersRouter = OneOf {
    Route(.case(UsersRoute.user)) {
        Path { Parse(.string) }
        userRouter
    }
    
    Route(.case(UsersRoute.update)) {
        Method.put
        Body(.json(UserOutput.self))
    }
}

public enum AuthEngineRoute: Equatable {
    case users(UsersRoute)
   ........
}

public let authEngineRoute = OneOf {
    Route(.case(AuthEngineRoute.users)) {
        Path { "users" }
        usersRouter
    }
    ......
}


```

calling from iOS app this 

```swift
return try await api.apiClient.decodedResponse(
            for: .authEngine(.users(.user(id: id, route: .delete))),
            as: DeleteResponse.self,
            decoder: .ISO8601JSONDecoder
          ).value
``` but always send HTTP GET method why?

**Comment by stephencelis** [comment]
@saroar Do you have a repro we could check out to investigate? We have some rudimentary tests on the URL router that the method should be preserved and have found it to work in practice for us, but maybe there's something breaking in the composition/API client.

**Reply by saroar** [reply]
yes I think I found issue 
before I have 
4 VAPOR 4 APP

> API_GATEWAY_SERVER_APP(0.0.0.0:8080) this app was handle all request other engine 
https://github.com/AddaMeSPB/AddaAPIGateway
SiteRouter + Models
https://github.com/AddaMeSPB/adda-shared-models

1.   > API_AUTH_ENGINE_SERVER_APP(0.0.0.0:9090) https://github.com/AddaMeSPB/AddaMeAuth/tree/siteRouter
2.   > API_EVENT_ENGINE_SERVER_APP(0.0.0.0:3030) https://github.com/AddaMeSPB/EventEngine
3.   > API_CHAT_ENGINE_SERVER_APP(0.0.0.0:6060) https://github.com/AddaMeSPB/ChatEngine/tree/siteRoute

1, 2, 3 app port ip was block from outside world only open port was 8080 
and I use Docker for publish my app 
but I am having issue how I will handle my SiteRoter from `API_GATEWAY_SERVER_APP`

and was issue was I was mismatch port in my iOS so it will was working 

it look like I have Microservices all Microservices(handle by gateway app)
can you give me some advice now how I handle this Microservices via handle API_GATEWAY_SERVER_APP so my API_GATEWAY_SERVER_APP have to add SiteRouter functions 

HOPE I dont confused you 
you can check how my microservices architecture are build 
https://rynaardb.com/posts/2019/02/23/microservices-with-vapor.html
